search-service/src/main/java/com/movie/recommender/search/service/SearchService.java

```diff
@@ -0,0 +1,83 @@
+package com.movie.recommender.search.service;
+
+import com.movie.recommender.common.model.movie.MovieDetails;
+import com.movie.recommender.search.model.SearchFilter;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.data.mongodb.core.MongoTemplate;
+import org.springframework.data.mongodb.core.query.Criteria;
+import org.springframework.data.mongodb.core.query.Query;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Slf4j
+@Service
+public class SearchService {
+    final private MongoTemplate mongoTemplate;
+
+    public SearchService(MongoTemplate mongoTemplate) {
+        this.mongoTemplate = mongoTemplate;
+    }
+
+    public List<MovieDetails> search(SearchFilter filter) {
```

Comment:
mongo doesn’t require the additional validation, as it will handle invalid or missing filters
keep validation only for enforcing business rules, as invalid inputs won’t cause significant issues
I would keep only simple checks 
```
if (filter.getMovieTitle() != null)
...
if (filter.getGenres() != null && !filter.getGenres().isEmpty()) 
...
if (filter.getStartDate() != null && filter.getEndDate() != null)
...
...
```